python-squid/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -95,6 +97,37 @@ public void visitLambda(PyLambdaExpressionTree pyLambdaExpressionTree) {
       super.scopeRootTrees.pop();
     }
 
+    @Override
+    public void visitFunctionDef(PyFunctionDefTree pyFunctionDefTree) {
+      createScope(pyFunctionDefTree, currentScope());
+      enterScope(pyFunctionDefTree);
+      createParameters(pyFunctionDefTree.parameters());
+      super.visitFunctionDef(pyFunctionDefTree);
+      super.scopeRootTrees.pop();
+    }
+
+    @Override
+    public void visitForStatement(PyForStatementTree pyForStatementTree) {
+      createLoopVariables(pyForStatementTree);
+      super.visitForStatement(pyForStatementTree);
+    }
+
+    @Override
+    public void visitComprehensionFor(PyComprehensionForTree tree) {
+      if (tree.loopExpression().is(Tree.Kind.NAME)) {
+        addUsage((PyNameTree) tree.loopExpression());
+      }
+      super.visitComprehensionFor(tree);
+    }
+
+    private void createLoopVariables(PyForStatementTree loopTree) {
+      loopTree.expressions().forEach(expr -> {
+        if (expr.is(Tree.Kind.NAME)) {
```

Comment:
What about tuples?

Reply:
I will create a separate ticket to handle tuples